### PR TITLE
Switch-Azure mode - added RM profile project

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

File: src/Common/Commands.ResourceManager.Profile/Microsoft.Azure.Commands.Profile.dll-Help.psd1 [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+GUID = 'D48CF693-4125-4D2D-8790-1514F44CE325'
+
+# Author of this module
+Author = 'Microsoft Corporation'
+
+# Company or vendor of this module
+CompanyName = 'Microsoft Corporation'
+
+# Copyright statement for this module
+Copyright = '?Microsoft Corporation. All rights reserved.'
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Microsoft.Azure.Commands.Profile.dll-Help.psd1]: Nit: use existing psd1 file to align versions, copyright etc...

File: src/Common/Commands.ResourceManager.Profile/Microsoft.Azure.Commands.Profile.format.ps1xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Microsoft.Azure.Commands.Profile.format.ps1xml]: We dont need this

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public class SaveAzureProfileCommand : AzurePSCmdlet
+    {
+        [Parameter(Mandatory = false, Position = 0, ValueFromPipelineByPropertyName = true)]
+        public AzureRMProfile Profile { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true)]
+        public string Path { get; set; }
+
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        public void ExecuteCmdlet()
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: use `ProcessRecord` pattern instead

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.Profile.Properties;
+
+namespace Microsoft.Azure.Commands.Profile
+{
+    /// <summary>
+    /// Saves Microsoft Azure profile.
+    /// </summary>
+    [Cmdlet(VerbsData.Save, "AzureProfile"), OutputType(typeof(AzureSMProfile))]
+    public class SaveAzureProfileCommand : AzurePSCmdlet
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: Extend from `AzureRMCmdlet`

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                    throw new ArgumentException(Resources.AzureProfileMustNotBeNull);
+                }
+                AzureRMCmdlet.Profile.Save(Path);
+            }
+
+            WriteVerbose(string.Format("Profile saved to: {0}.", Path));
+        }
+
+        protected override AzureContext DefaultContext
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: Remove

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+        protected override void ProcessRecord()
+        {
+            try
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: Remove try/catch code

File: src/Common/Commands.ResourceManager.Profile/Profile/SelectAzureProfile.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Management.Automation;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.Profile.Properties;
+
+namespace Microsoft.Azure.Commands.Profile
+{
+    /// <summary>
+    /// Creates new Microsoft Azure profile.
```
Comment (ogail) [inline src/Common/Commands.ResourceManager.Profile/Profile/SelectAzureProfile.cs]: selects new ....

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs (hunk 3 above)
Comment (ljhljh235) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: addressed

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs (hunk 4 above)
Comment (ljhljh235) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: addressed

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs (hunk 5 above)
Comment (ljhljh235) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: addressed

File: src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs (hunk 6 above)
Comment (ljhljh235) [inline src/Common/Commands.ResourceManager.Profile/Profile/SaveAzureProfile.cs]: addressed

File: src/Common/Commands.ResourceManager.Profile/Profile/SelectAzureProfile.cs (hunk 7 above)
Comment (ljhljh235) [inline src/Common/Commands.ResourceManager.Profile/Profile/SelectAzureProfile.cs]: addressed

**Pull request merged**